Fixes for Coverity warnings around ostree_repo_remote_get_url

debarshiray opened the pull request:
Going through my downstream Coverity reports, I see that these are relevant for `main` as it stands today.

---

* dir, system-helper: Don't ignore errors when getting a remote's URL

* system-helper: Improve the errors on failure to get a remote's URL

debarshiray commented:
I wonder why the *Build with gcc and test (older)* test seems to be stuck.  Let's see if I have better luck, if I trigger it again with another force-push.

debarshiray commented:
The `tests/test-run@system-norevokefs,nodeltas.wrap` test is failing on Ubuntu 18.04 when built with GCC, libsoup, and the builtin bubblewrap and xdg-dbus-proxy.

However, it passes on my Fedora 35 when built the same way.

debarshiray commented:
> The `tests/test-run@system-norevokefs,nodeltas.wrap` test is failing

Looks like a force-push got it going.  For posterity, the test logs had:
```
++ run_sh org.test.Hello readlink /proc/self/ns/net
++ ID=org.test.Hello
++ shift
++ flatpak run --command=bash org.test.Hello -c 'readlink /proc/self/ns/net'
bwrap: Can't bind mount /oldroot/tmp/test-flatpak-vxR0xN/system/runtime/org.test.Platform/x86_64/stable/3fd980bdedb381088b60707ac421df6c514098217f421ac71164e3bb6ec638a4/files on /newroot/usr: Unable to find destination in mount table
+ cleanup
+ /bin/kill -9 6513
+ gpg-connect-agent --homedir /tmp/test-flatpak-vxR0xN/gpghome killagent /bye
OK closing connection
+ fusermount -u /tmp/test-flatpak-vxR0xN/runtime/doc
fusermount: entry for /tmp/test-flatpak-vxR0xN/runtime/doc not found in /etc/mtab
```

debarshiray commented:
Damn, the Valgrind job got cancelled midway through the `make distcheck` step because it took longer than 6 hours.  The other jobs don't do a `distcheck`, but they do run `make check` and those take less than 10 minutes.

Maybe we just hit a slow host?  Let's try again.

debarshiray commented:
Let's continue to discuss this in https://github.com/flatpak/flatpak/pull/4948

mwleeds reviewed (commented):
I didn't review all the commits here yet, just the first two

alexlarsson commented:
lgtm other than the existing comments

debarshiray commented on app/flatpak-builtins-utils.c:
```diff
@@ -1401,6 +1401,9 @@ ensure_remote_state_arch_for_ref (FlatpakDir         *dir,
 {
   g_autofree char *ref_arch = flatpak_get_arch_for_ref (ref);
+  if (ref_arch == NULL)
+    return flatpak_fail_error (error, FLATPAK_ERROR_INVALID_REF, "Unable to get arch for %s", ref);
```
I went with `g_return_val_if_fail()` but I am happy to change it.

debarshiray commented:
Rebased against `main`.  My apologies for the delay.

I added one commit to fix the order of architecture and branch in triplets in comments.  After staring long enough at the ref completion code, the wrong order started to bother me.  :)

debarshiray commented on common/flatpak-installation.c:
Done

debarshiray commented on common/flatpak-installation.c:
```diff
@@ -3123,9 +3123,13 @@ flatpak_installation_list_unused_refs_with_options (FlatpakInstallation *self,
       g_autoptr(GError) local_error = NULL;
       FlatpakInstalledRef *ref = NULL;
       g_autoptr(FlatpakDecomposed) decomposed = flatpak_decomposed_new_from_ref (*iter, &local_error);
```
Done
